src/backend/distributed/citus--8.0-12--8.0-13.sql

```diff
@@ -0,0 +1,33 @@
+/* citus--8.0-12--8.0-13 */
+CREATE FUNCTION citus_reset_default_for_node_conninfo()
+    RETURNS void
+    LANGUAGE C STRICT
+    AS 'MODULE_PATHNAME', $$citus_reset_default_for_node_conninfo$$;
+
+DO LANGUAGE plpgsql
+$$
+BEGIN
+    -- citus requires ssl form this version onwards. For older installations that didn't
+    -- change the citus.node_conninfo setting this will cause the new default value to be
+    -- used.
+    -- This default value is incompatible with postgres' default setting for ssl. When we
```

Comment:
I agree that the comment needs a revision.

Reading your suggestion I think some more clarification about the whole change is in needed.

When existing users upgrade to 8.1 we will specifically **not** require the use of ssl and default back to `sslmode=prefer`.

Given that the default is changed in the code it will impact users that didn't specifically set `citus.node_conninfo`. Since no configuration line is in `postgresql.conf` in such a case the default from the code will be used.

This code tests for this case and reverts back the `citus.node_conninfo` to its default value pre 8.1 by adding it to `postgres.auto.conf` via `ALTER SYSTEM`.

I'll take your comment as the basis for a revision here.